indexer/bag_indexer/src/main/scala/uk/ac/wellcome/platform/archive/indexer/bags/services/FileSuffixCounter.scala

```diff
@@ -0,0 +1,14 @@
+package uk.ac.wellcome.platform.archive.indexer.bags.services
+
+import uk.ac.wellcome.platform.archive.common.storage.models.StorageManifestFile
+
+object FileSuffixCounter {
+  def count(files: Seq[StorageManifestFile]): Map[String, Int] = {
+    files
+      .map(_.name.split("\\."))
+      .filter(_.length > 1)
+      .map(_.reverse.head)
+      .groupBy(identity)
+      .mapValues(_.size)
+  }
```

Comment:
😻 

FP style makes this very nice.

Two suggestions:

* Can you replace `.map(_.reverse.head)` with `.map(_.last)`?
* Should we lowercase the suffix? e.g. if a bag has `cat.jp2` and `dog.JP2`, we should combine those together into a single entry.

Reply:
Me while writing this "I wish I could just get the last element of a list".

Reply:
Lowercasing is a definite yes.